src/properties.jl

```diff
@@ -0,0 +1,668 @@
+function libver_bound_to_enum(val::VersionNumber)
+    val >= v"1.12"   ? API.H5F_LIBVER_V112 :
+    val >= v"1.10"   ? API.H5F_LIBVER_V110 :
+    val >= v"1.8"    ? API.H5F_LIBVER_V18 :
+    throw(ArgumentError("libver_bound must be >= v\"1.8\"."))
+end
+function libver_bound_to_enum(val::Symbol)
+    val == :earliest ? API.H5F_LIBVER_EARLIEST :
+    val == :latest   ? libver_bound_to_enum(libversion) :
+    throw(ArgumentError("Invalid libver_bound $val."))
+end
+function libver_bound_from_enum(enum)
+    enum == API.H5F_LIBVER_EARLIEST ? :earliest :
+    enum == API.H5F_LIBVER_V18      ? v"1.8" :
+    enum == API.H5F_LIBVER_V110     ? v"1.10" :
+    enum == API.H5F_LIBVER_V112     ? v"1.12" :
+    error("Unknown libver_bound value $enum")
+end
+function get_libver_bounds(p::Properties)
+    low, high = API.h5p_get_libver_bounds(p)
+    return libver_bound_from_enum(low), libver_bound_from_enum(high)
+end
+function set_libver_bounds!(p::Properties, (low, high)::Tuple{Any,Any})
+    API.h5p_set_libver_bounds(p, libver_bound_to_enum(low), libver_bound_to_enum(high))
+end
+function set_libver_bounds!(p::Properties, val)
+    API.h5p_set_libver_bounds(p, libver_bound_to_enum(val), libver_bound_to_enum(val))
+end
+
+
+class_propertynames(::Type{FileAccessProperties}) = (
+    :alignment,
+    :driver,
+    :driver_info,
+    :fapl_mpio,
+    :fclose_degree,
+    :libver_bounds,
+    )
+
+function class_getproperty(::Type{FileAccessProperties}, p::Properties, name::Symbol)
+    name === :alignment     ? get_alignment(p) :
+    name === :driver        ? Drivers.get_driver(p) :
+    name === :driver_info   ? API.h5p_get_driver_info(p) : # get only
+    name === :fclose_degree ? get_fclose_degree(p) :
+    name === :libver_bounds ? get_libver_bounds(p) :
+    # deprecated
+    name === :fapl_mpio     ? (depwarn("The `fapl_mpio` property is deprecated, use `driver=HDF5.Drivers.MPIO(...)` instead.", :fapl_mpio); drv = get_driver(p, MPIO); (drv.comm, drv.info)) :
+    class_getproperty(superclass(FileAccessProperties), p, name)
+end
+function class_setproperty!(::Type{FileAccessProperties}, p::Properties, name::Symbol, val)
+    name === :alignment     ? set_alignment!(p, val) :
+    name === :driver        ? Drivers.set_driver!(p, val) :
+    name === :fclose_degree ? set_fclose_degree!(p, val) :
+    name === :libver_bounds ? set_libver_bounds!(p, val) :
+    # deprecated
+    name === :fapl_mpio     ? (depwarn("The `fapl_mpio` property is deprecated, use `driver=HDF5.Drivers.MPIO(...)` instead.", :fapl_mpio); p.driver = Drivers.MPIO(val...)) :
+    class_setproperty!(superclass(FileAccessProperties), p, name, val)
+end
+
+
+@propertyclass LinkAccessProperties API.H5P_LINK_ACCESS
+
+@propertyclass GroupAccessProperties API.H5P_GROUP_ACCESS
+superclass(::Type{GroupAccessProperties}) = LinkAccessProperties
+
+@propertyclass DatatypeAccessProperties API.H5P_DATATYPE_ACCESS
+superclass(::Type{DatatypeAccessProperties}) = LinkAccessProperties
```

Comment:
What was the reason that these all use `LinkAccessProperties` as their superclass?

Reply:
I think I queried them via `H5P_GET_CLASS_PARENT` to see what they did internally. Would probably make sense to add that as a test.